### PR TITLE
feat: added Query func to run generic SQL queries

### PR DESCRIPTION
**Pull request by adityathebe**

### CODE REVIEW

File: query.go [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+	"database/sql"
+	"fmt"
+
+	"github.com/flanksource/commons/logger"
+)
+
+// Query runs the given SQL query against the provided db connection.
+// The rows are returned as a map of columnName=>columnValue.
+func Query(ctx context.Context, conn *sql.DB, query string) ([]map[string]any, error) {
+	tx, err := conn.BeginTx(ctx, &sql.TxOptions{ReadOnly: true})
```
Comment (moshloop) [inline query.go:14]: I think use dbpool *pgxpool.Pool, to make it consistent

Comment (moshloop) [inline query.go:14]: I don't think we need to start and commit a transaction for a readonly query

Comment (adityathebe) [reply query.go]: That's just to ensure that only read queries are run.

File: query.go [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+		results = append(results, row)
+	}
+
+	err = rows.Err()
+	if err != nil {
+		return nil, err
+	}
+
+	return results, nil
```
Comment (moshloop) [inline query.go:52]: nit: You could do

```golang

if err := rows.Err(); err != nil {
return nil, err
}
```

or even just `return results, rows.Err()`

**Pull request merged**